File: Axiom/Internal/Ast/Identifier.cs
namespace Axiom.Internal.Ast
{
    internal class Identifier : Expression
    {
        public Identifier(string name)
        {
            Name = name;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public override int GetHashCode()
        {
            return Name.GetHashCode();
        }

        public override bool Equals(object obj)
        {
            return Name == ((Identifier)obj).Name;
        }

        public static bool operator ==(Identifier left, Identifier right) {
            return left.Equals(right);
        }

        public static bool operator !=(Identifier left, Identifier right)
        {
            return !left.Equals(right);
        }

        public string Name { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/Identifier.java
package com.moybl.axiom.ast;

import com.moybl.axiom.Position;

public class Identifier extends Expression {

	private String name;

	public Identifier(Position position, String name) {
		super(position);
		this.name = name;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	@Override
	public boolean equals(Object obj) {
		if (obj == null || obj.getClass() != this.getClass()) {
			return false;
		}

		return name.equals(((Identifier) obj).name);
	}

	@Override
	public int hashCode() {
		return name.hashCode();
	}

	public String getName() {
		return name;
	}

}
